apps/etterlatte-saksbehandling-ui/client/src/components/behandling/aktivitetsplikt/AktivitetspliktTidslinje.tsx

```diff
@@ -10,54 +10,61 @@ import {
 } from '@navikt/aksel-icons'
 import { BodyShort, Button, HStack, Timeline, ToggleGroup, VStack } from '@navikt/ds-react'
 import {
-  hentAktiviteterForBehandling,
-  hentAktiviteterForSak,
+  hentAktiviteterOgHendelserForBehandling,
+  hentAktiviteterOgHendelserForSak,
   slettAktivitet,
   slettAktivitetForSak,
+  slettAktivitetHendelse,
+  slettAktivitetHendelseForSak,
 } from '~shared/api/aktivitetsplikt'
 import { formaterDato, formaterDatoMedTidspunkt } from '~utils/formatering/dato'
 import { IDetaljertBehandling } from '~shared/types/IDetaljertBehandling'
 import { addMonths, addYears } from 'date-fns'
 import { useApiCall } from '~shared/hooks/useApiCall'
 import React, { ReactNode, useEffect, useState } from 'react'
-import { AktivitetspliktType, IAktivitet } from '~shared/types/Aktivitetsplikt'
+import { AktivitetspliktType, IAktivitetHendelse, IAktivitetPeriode } from '~shared/types/Aktivitetsplikt'
 import { isFailureHandler } from '~shared/api/IsFailureHandler'
-import { NyAktivitet } from '~components/behandling/aktivitetsplikt/NyAktivitet'
 import { isPending } from '~shared/api/apiUtils'
 import Spinner from '~shared/Spinner'
+import { NyAktivitetHendelse } from '~components/behandling/aktivitetsplikt/AktivitetOgHendelse'
 
 interface Props {
   behandling?: IDetaljertBehandling
   doedsdato: Date
-  sakId?: number
+  sakId: number
 }
 
 export const AktivitetspliktTidslinje = ({ behandling, doedsdato, sakId }: Props) => {
-  const [hentet, hent] = useApiCall(hentAktiviteterForBehandling)
-  const [hentetForSak, hentForSak] = useApiCall(hentAktiviteterForSak)
+  const [hentet, hent] = useApiCall(hentAktiviteterOgHendelserForBehandling)
+  const [hentetForSak, hentForSak] = useApiCall(hentAktiviteterOgHendelserForSak)
   const [slettet, slett] = useApiCall(slettAktivitet)
+  const [slettetHendelse, slettHendelse] = useApiCall(slettAktivitetHendelse)
   const [slettetForSak, slettForSak] = useApiCall(slettAktivitetForSak)
+  const [slettetHendelseForSak, slettHendelseForSak] = useApiCall(slettAktivitetHendelseForSak)
   const seksMndEtterDoedsfall = addMonths(doedsdato, 6)
   const tolvMndEtterDoedsfall = addMonths(doedsdato, 12)
-
-  const [aktiviteter, setAktiviteter] = useState<IAktivitet[]>([])
-  const [rediger, setRediger] = useState<IAktivitet | undefined>(undefined)
+  const [aktiviteter, setAktiviteter] = useState<IAktivitetPeriode[]>([])
+  const [hendelser, setHendelser] = useState<IAktivitetHendelse[]>([])
+  const [rediger, setRediger] = useState<IAktivitetPeriode | undefined>(undefined)
+  const [redigerHendelse, setRedigerHendelse] = useState<IAktivitetHendelse | undefined>(undefined)
   const [aktivitetsTypeProps, setAktivitetsTypeProps] = useState<AktivitetstypeProps[]>([])
   const [sluttdato, setSluttdato] = useState<Date>(addYears(doedsdato, 3))
```

Comment:
Her er det veldig mye state. Samtidig som veldig mye av det har også lite beskrivende navn for hva som som holdes i staten. 

Her luktes det som at ting kan splittes opp i mer hensiksmessige komponenter

Reply:
API state følger også ikke navngivning som resten av koden til Gjenny har